src/lib/statistics/chunk_statistics/histograms/equal_width_histogram.hpp

```diff
@@ -0,0 +1,187 @@
+#pragma once
+
+#include <memory>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "abstract_histogram.hpp"
+#include "histogram_utils.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename T>
+struct EqualWidthBinData {
+  T minimum;
+  T maximum;
+  std::vector<HistogramCountType> bin_counts;
+  std::vector<HistogramCountType> bin_distinct_counts;
+  BinID bin_count_with_larger_range;
+};
+
+/**
+ * Width-balanced histogram.
+ * Bins are of roughly equal width, i.e., cover the same share of the domain of values.
+ * Bins are consecutive (no gaps between bins), but might be empty.
+ */
+template <typename T>
+class EqualWidthHistogram : public AbstractHistogram<T> {
+ public:
+  using AbstractHistogram<T>::AbstractHistogram;
+
+  EqualWidthHistogram(const T minimum, const T maximum, const std::vector<HistogramCountType>& bin_heights,
+                      const std::vector<HistogramCountType>& bin_distinct_counts,
+                      const BinID bin_count_with_larger_range);
+  EqualWidthHistogram(const std::string& minimum, const std::string& maximum,
+                      const std::vector<HistogramCountType>& bin_heights,
+                      const std::vector<HistogramCountType>& bin_distinct_counts,
+                      const BinID bin_count_with_larger_range, const std::string& supported_characters,
+                      const uint32_t string_prefix_length);
+
+  /**
+   * Create a histogram based on the data in a given segment.
+   * @param segment The segment containing the data.
+   * @param max_bin_count The number of bins to create. The histogram might create fewer, but never more.
+   * @param supported_characters A sorted, consecutive string of characters supported in case of string histograms.
+   * Can be omitted and will be filled with default value.
+   * @param string_prefix_length The prefix length used to calculate string ranges.
+   * * Can be omitted and will be filled with default value.
+   */
+  static std::shared_ptr<EqualWidthHistogram<T>> from_segment(
+      const std::shared_ptr<const BaseSegment>& segment, const BinID max_bin_count,
+      const std::optional<std::string>& supported_characters = std::nullopt,
+      const std::optional<uint32_t>& string_prefix_length = std::nullopt);
+
+  HistogramType histogram_type() const override;
+  std::string histogram_name() const override;
+  HistogramCountType total_distinct_count() const override;
+  HistogramCountType total_count() const override;
+
+  /**
```

Comment:
This is the same comment as on `AbstractHistogram::bin_count()`, right? No need to have this one here as well then, they will just desync and confuse people.

Reply:
It's not the same comment. The one in `AbstractHistogram` refers to the comments in the specific histogram types to explain why and how the actual number of bins can differ from the ones requested during generation.